doozerlib/metadata.py

```diff
@@ -229,102 +405,137 @@ def get_component_name(self, default=-1) -> str:
+        Returns: (rebuild:<bool>, message: description of why).
         """
+        now = datetime.datetime.utcnow()
+
         component_name = self.get_component_name(default='')
         if not component_name:
             # This can happen for RPMs if they have never been rebased into
             # distgit.
-            return True, 'Could not find component name; assuming never built'
+            return RebuildHint(code=RebuildHintCode.NO_COMPONENT,
+                               reason='Could not determine component name; assuming it has never been built')
 
-        # latest_build_creation_event_id = latest_build['creation_event_id']
-        # all candidate Brew tags configured for this component. e.g. [rhaos-4.7-rhel-8-candidate, rhaos-4.7-rhel-7-candidate]
-        candidate_tags = self.candidate_brew_tags()
-        with self.runtime.pooled_koji_client_session() as koji_api:
-            # latest builds of this component in all candidate Brew tags (e.g. [rhaos-4.7-rhel-8-candidate, rhaos-4.7-rhel-7-candidate])
-            build_lists = brew.get_latest_builds([(tag, component_name) for tag in candidate_tags], None, None, koji_api)
-            latest_builds = [builds[0] if builds else None for builds in build_lists]
-            tags_without_builds = {tag for tag, _ in filter(lambda tag_build: tag_build[1] is None, zip(candidate_tags, latest_builds))}
-            if tags_without_builds:
-                return True, f'Component {component_name} has never been built against {tags_without_builds}'
-
-            # latest_build is the eldest among those latest builds for different targets
-            latest_build = min(latest_builds, key=lambda build: build['creation_event_id'])
-            latest_build_creation_event_id = latest_build['creation_event_id']
-            # getEvent returns something like {'id': 31825801, 'ts': 1591039601.2667}
-            latest_build_creation_ts_seconds = int(koji_api.getEvent(latest_build_creation_event_id)['ts'])
-            # Log scan-sources coordinates throughout to simplify setting up scan-sources
-            # function tests to reproduce real-life scenarios.
-            self.logger.debug(f'scan-sources coordinate: latest_build: {latest_build}')
-            self.logger.debug(f'scan-sources coordinate: latest_build_creation_ts_seconds: {latest_build_creation_ts_seconds}')
-
-        dgr = self.distgit_repo()
-        with Dir(dgr.distgit_dir):
-            dg_commit, _ = exectools.cmd_assert('git rev-parse HEAD', strip=True)
-            self.logger.debug(f'scan-sources coordinate: dg_commit: {dg_commit}')
-            ts, _ = exectools.cmd_assert('git show -s --format=%ct HEAD', strip=True)
-            distgit_head_commit_ts_seconds = int(ts)
-            self.logger.debug(f'scan-sources coordinate: distgit_head_commit_ts_seconds: {distgit_head_commit_ts_seconds}')
+        latest_build = self.get_latest_build(default=None)
```

Comment:
For rpms, we are building for both -el7 and -el8 tags. Previously, we were getting the latest builds for each tag, then use the older one. So if the latest builds for el7 and el8 don't match, it will request a rebuild. This PR removes that mechanism.

Reply:
Nice catch! Added extra internal `get_latest_targeted_build` to handle this.

Reply:
Nice catch! `needs_rebuild` will now check both el7/el8 targets if they are defined in the rpm meta.